typings/index.d.ts

```diff
@@ -173,9 +173,9 @@ export class BaseClient extends EventEmitter {
   public clearTimeout(timeout: NodeJS.Timeout): void;
   public clearImmediate(timeout: NodeJS.Immediate): void;
   public destroy(): void;
-  public setInterval(fn: (...args: unknown[]) => void, delay: number, ...args: unknown[]): NodeJS.Timeout;
-  public setTimeout(fn: (...args: unknown[]) => void, delay: number, ...args: unknown[]): NodeJS.Timeout;
-  public setImmediate(fn: (...args: unknown[]) => void, ...args: unknown[]): NodeJS.Immediate;
+  public setInterval<T extends any[]>(fn: (...args: T) => Awaited<void>, delay: number, ...args: T): NodeJS.Timeout;
```

Comment:
If you replace the T with `T extends unknown[]`.. don't you achieve the same? since TS will (hopefully) infer it

Reply:
I’m not sure since I’m not experienced with this syntax so I’ll tag @vaporox and @1Computer1